plugins/optimization-detective/class-od-strict-url-metric.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Optimization Detective: OD_Strict_URL_Metric class
+ *
+ * @package optimization-detective
+ * @since n.e.x.t
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Representation of the measurements taken from a single client's visit to a specific URL without additionalProperties allowed.
+ *
+ * This is used exclusively in the REST API endpoint for capturing new URL metrics to prevent invalid additional data from being
+ * submitted in the request. For URL metrics which have been stored the looser OD_URL_Metric class is used instead.
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+final class OD_Strict_URL_Metric extends OD_URL_Metric {
```

Comment:
It's a bit confusing to have "strict" URL metric extend the "extendable" URL metric - would make more sense to be the other way around. This is also highlighted by how the strict class will still run the filters, but without that serving any purpose.

Taking this a bit further, strictly speaking neither of the two makes sense because a strict URL metric is not an extendable URL metric and vice-versa. So it would make more sense to use e.g. a decorator pattern, or make the two classes separate implementations of a `OD_URL_Metric` interface. Of course that would be a decent amount of refactoring to do - simple, but still some code changes so we could consider that separately.

So at this point I mostly think:
* Should the extension tree be the other way around? If not, why not?
* Maybe there's a more clear name we can use than just "strict"? Maybe include "REST" so that it's clear that this class is only used for REST API requests?

Reply:
I'm not sure I understand. The filters are still used. It's just that any objects have `additionalProperties` set to `false` in the strict version.

Reply:
In other words, in the strict version, it sets `additionalProperties` to `false` which refers to the properties which aren't recognized as part of the schema. The filters are still applying, however, so if a plugin adds new properties to the schema these "additional" properties will be recognized. It's just that additional _unrecognized_ properties will be rejected in the strict verison.

Reply:
Ah right, I had something wrong in my logic here. Some of my other feedback on the class hierarchy still applies, but that's not a big deal in regards to this PR. I _do_ think we should separately consider using a `OD_URL_Metric` interface and use that in all (or at least most of) the type hints so that the class implementations can be independent.